src/models/quote.js

```diff
@@ -122,7 +125,26 @@ function * getAccountLedger (account) {
 
 function * getQuote (params, ledgers, config) {
   const query = yield makeQuoteQuery(params, config)
+  const type = query.source_amount ? 'source' : 'destination'
+  const amount = query.source_amount || query.destination_amount
   const quote = yield backend.getQuote(makeQuoteArgs(query))
+  try {
+    validate('Quote', quote)
```

Comment:
What about adding an `isValid` method to the validator? That way you don't have try/catch/rethrow